src/block/tile/Sign.php

```diff
@@ -77,22 +77,30 @@ public function __construct(World $world, Vector3 $pos){
 		parent::__construct($world, $pos);
 	}
 
+	private function readTextTag(CompoundTag $nbt, bool $lightingBugResolved) : void{
+		$baseColor = new Color(0, 0, 0);
+		$glowingText = false;
+		if(($baseColorTag = $nbt->getTag(self::TAG_TEXT_COLOR)) instanceof IntTag){
+			$baseColor = Color::fromARGB(Binary::unsignInt($baseColorTag->getValue()));
+		}
+		if($lightingBugResolved && ($glowingTextTag = $nbt->getTag(self::TAG_GLOWING_TEXT)) instanceof ByteTag){
+			//both of these must be 1 - if only one is set, it's a leftover from 1.16.210 experimental features
+			//see https://bugs.mojang.com/browse/MCPE-117835
+			$glowingText = $glowingTextTag->getValue() !== 0;
+		}
+		$this->text = SignText::fromBlob(mb_scrub($nbt->getString(self::TAG_TEXT_BLOB), 'UTF-8'), $baseColor, $glowingText);
+	}
+
 	public function readSaveData(CompoundTag $nbt) : void{
-		if(($textBlobTag = $nbt->getTag(self::TAG_TEXT_BLOB)) instanceof StringTag){ //MCPE 1.2 save format
-			$baseColor = new Color(0, 0, 0);
-			$glowingText = false;
-			if(($baseColorTag = $nbt->getTag(self::TAG_TEXT_COLOR)) instanceof IntTag){
-				$baseColor = Color::fromARGB(Binary::unsignInt($baseColorTag->getValue()));
-			}
-			if(
-				($glowingTextTag = $nbt->getTag(self::TAG_GLOWING_TEXT)) instanceof ByteTag &&
-				($lightingBugResolvedTag = $nbt->getTag(self::TAG_LEGACY_BUG_RESOLVE)) instanceof ByteTag
-			){
-				//both of these must be 1 - if only one is set, it's a leftover from 1.16.210 experimental features
-				//see https://bugs.mojang.com/browse/MCPE-117835
-				$glowingText = $glowingTextTag->getValue() !== 0 && $lightingBugResolvedTag->getValue() !== 0;
-			}
-			$this->text = SignText::fromBlob(mb_scrub($textBlobTag->getValue(), 'UTF-8'), $baseColor, $glowingText);
+		$lightingBugResolved = false;
```

Comment:
This ought to be `true` by default

Reply:
Actually, you can just put this variable inside the TAG_TEXT_BLOB branch and pass `true` in the other branch